ls1intum/Artemis
# `Quiz exercises`: Improve quiz question points validation

### Checklist
#### General


#### Client

### Motivation and Context



When an instructor set the question point to 99999999999, the save button was enabled. After clicking the save button, there was Bad Request Error:
![image](https://user-images.githubusercontent.com/10885503/217834636-fbf05083-d466-4091-bc3d-6fd4e134dddc.png)

This error should be prevented by disabling the save button if the points are in invalid range.

### Description

According to the existing implementation in [multiple-choice-question-edit.component.html](https://github.com/ls1intum/Artemis/blob/c5e456e973d3c3145fc52520eb1b15eccaca8dc9/src/main/webapp/app/exercises/quiz/manage/multiple-choice-question/multiple-choice-question-edit.component.html#L23), [drag-and-drop-question-edit.component.html](https://github.com/ls1intum/Artemis/blob/c5e456e973d3c3145fc52520eb1b15eccaca8dc9/src/main/webapp/app/exercises/quiz/manage/drag-and-drop-question/drag-and-drop-question-edit.component.html#L23), and [short-answer-question-edit.component.html](https://github.com/ls1intum/Artemis/blob/c5e456e973d3c3145fc52520eb1b15eccaca8dc9/src/main/webapp/app/exercises/quiz/manage/short-answer-question/short-answer-question-edit.component.html#L23), the max points of each question should be 9999. Therefore, the condition to check if points are greater than 9999 is added to quiz-exercise-validation.directive.ts

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course

1. Log in to Artemis
2. Navigate to Course Administration
3. Select the Course
4. Navigate to Exercises
5. Click **+ Create Quiz** Button
6. Input Quiz Title
7. Add any type of question
8. Input Question Title
9. Verify that the **Save** Button is enabled and no **Invalid Input** sign
10. Modify the Points to 99999999999
11. Verify that the **Save** Button is disabled and there is **Invalid Input** sign

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| quiz-exercise-validation.directive.ts | 91.01% | 97.54% |

### Screenshots


<img width="1505" alt="Screenshot 2023-02-10 at 15 55 16" src="https://user-images.githubusercontent.com/10885503/218122919-f45cdc84-29eb-421f-a57f-24c991de8d7c.png">
